mlir/lib/Target/LLVMIR/ModuleTranslation.cpp

```diff
@@ -1030,10 +1030,15 @@ LogicalResult ModuleTranslation::convertGlobals() {
       llvm::DIGlobalVariable *diGlobalVar = diGlobalExpr->getVariable();
       var->addDebugInfo(diGlobalExpr);
 
+      // For fortran, the scope hierarchy can be
+      // variable -> module -> compile unit
+      llvm::DIScope *scope = diGlobalVar->getScope();
+      if (llvm::DIModule *mod = dyn_cast_if_present<llvm::DIModule>(scope))
+        scope = mod->getScope();
+
```

Comment:
Changes to MLIR should go in as a separate patch. And a test is required for the MLIR changes in I guess `mlir/test/Target/LLVMIR/llvmir-debug.mlir`.

Reply:
Thanks for letting me know. I have now opened #91604 with this change and a testcase. I will remove this bit from this PR.